build/ci_utils/src/actions/artifacts.rs

```diff
@@ -138,14 +138,49 @@ pub fn single_file_provider(
     Ok(futures::stream::iter([file]))
 }
 
+/// Scan recursively directory with its subtree for files to upload to the artifact.
+///
+/// The directory name will be preserved in the artifact container.
+///
+/// # Example
+/// ```
+/// # use ide_ci::prelude::*;
+/// # use tempfile::TempDir;
+/// # use ide_ci::actions::artifacts::single_dir_provider;
+/// # use std::path::Path;
+///
+/// # #[tokio::main]
+/// # async fn main() -> Result {
+/// let dir = TempDir::new()?;
+/// let dir = dir.path().join("uploaded_dir");
+/// let file1 = dir.join("file");
```

Comment:
You can consider adding a sibling of `uploaded_dir` to make sure it is not uploaded.